### PR TITLE
Test/improvements

### PR DESCRIPTION
**Pull request by davidseid**
@tremarkley -- Hey Harry, new pull request for you. Please take a look when you get a moment. This PR concerns some added unit tests, mainly testing a few more user interactions to improve my test coverage on the components. Still some gaps for sure as I ran into some issues testing state changes and asynchronous functions. Thanks!

### CODE REVIEW

**Comment by davidseid** [comment]
[Test conditional rendering in the thumbnail component](https://trello.com/c/yZQn9qVk/94-test-conditional-rendering-in-the-thumbnail-component)

**Comment by davidseid** [comment]
[Test toggleGalleryView in banner component](https://trello.com/c/vAyhQqX7/93-test-togglegalleryview-in-banner-component)

**Comment by davidseid** [comment]
[S - Set up test coverage tracking](https://trello.com/c/5m3tSTzD/92-s-set-up-test-coverage-tracking)

File: client/tests/Banner.test.js [hunk 1]
```diff
@@ -18,4 +19,12 @@ describe('Unit test for <Banner /> Component', () => {
     const wrapper = shallow(<Banner />);
     expect(wrapper.find('button').text()).toEqual('View Photos');
   });
+
+  it('should invoke toggleGalleryView on banner click', () => {
+    const toggleGalleryView = sinon.spy();
```
Comment (tremarkley) [inline client/tests/Banner.test.js:24]: If you would like to limit your dependencies you can also monitor whether a function has been called using a mock function in Jest: https://facebook.github.io/jest/docs/en/mock-functions.html

File: client/tests/Banner.test.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import Banner from '../src/Banner';
 import { shallow } from 'enzyme';
+import sinon from 'sinon';
 
 describe('Unit test for <Banner /> Component', () => {
   it('should render a banner with class name "banner"', () => {
```
Comment (tremarkley) [inline client/tests/Banner.test.js:7]: One small thing, you may have labelled this test incorrectly. It's description says it should be testing a class name but it looks like it is texting the text of the button.

File: client/tests/Spotlight.test.js [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import Spotlight from '../src/Spotlight';
 import { shallow } from 'enzyme';
+import sinon from "sinon";
 
 describe('Unit Test for <Spotlight /> Component', () => {
   it('should render a div with class name "spotlight"', () => {
```
Comment (tremarkley) [inline client/tests/Spotlight.test.js:7]: You should consider changing the description for the test. It seems to be testing a prop.

File: client/tests/Spotlight.test.js [hunk 4]
```diff
@@ -13,4 +14,28 @@ describe('Unit Test for <Spotlight /> Component', () => {
     const component = (<Spotlight numPhotos={13} />);
     expect(component.props.numPhotos).toEqual(13);
   });
+  
+  it('should invoke shiftSlide on user click of spotlight image', () => {
+    const testPhoto = {photoId: 2, url: 'testUrl', caption: 'testCaption'};
+    const shiftSlide = sinon.spy();
```
Comment (tremarkley) [inline client/tests/Spotlight.test.js:20]: If you would like to limit your dependencies you can also monitor whether a function has been called using a mock function in Jest: https://facebook.github.io/jest/docs/en/mock-functions.html

File: client/tests/Thumbnail.test.js [hunk 5]
```diff
@@ -18,4 +19,12 @@ describe('Unit Test for <Thumbnail /> Component', () => {
     const component = (<Thumbnail photoId={15} />);
     expect(component.props.photoId).toEqual(15);
   });
+
+  it('should call makeCurrentSlide on click of thumbnail', () => {
+    const makeCurrentSlide = sinon.spy();
```
Comment (tremarkley) [inline client/tests/Thumbnail.test.js:24]: If you would like to limit your dependencies you can also monitor whether a function has been called using a mock function in Jest: https://facebook.github.io/jest/docs/en/mock-functions.html

**Review by tremarkley — COMMENTED** [review]
Hey David, overall good job on this just a couple minor suggestions which you can find in my comments.

File: client/tests/Spotlight.test.js (hunk 3 above)
Comment (davidseid) [reply client/tests/Spotlight.test.js:7]: @tremarkley - I'm unsure about the test you are referring to here. Is it the test on line 7, 'should render a div with class name "spotlight"'? I thought in using expect and hasClass I would be testing classname and not prop, but let me know if I'm misunderstanding something here, thanks!

**Comment by davidseid** [comment]
@tremarkley -- Thanks for the feedback, Harry. I replaced my use of sinon with jest.fn() mock functions, thanks for the tip! Will go ahead and merge now.

**Pull request merged**